--- src/OptionHandlerFactory.cc.orig    2023-11-15 10:49:41.000000000 +0100
+++ src/OptionHandlerFactory.cc 2023-11-29 18:14:12.991687306 +0100
@@ -440,7 +440,7 @@
   {
     OptionHandler* op(new NumberOptionHandler(PREF_MAX_CONNECTION_PER_SERVER,
                                               TEXT_MAX_CONNECTION_PER_SERVER,
-                                              "1", 1, 16, 'x'));
+                                              "1", 1, 128, 'x'));
     op->addTag(TAG_BASIC);
     op->addTag(TAG_FTP);
     op->addTag(TAG_HTTP);

